### PR TITLE
[Frontend] Fix the TFMA/TFDV viz behavior when caching is used.

### PR DESCRIPTION
**Pull request by numerology**
Reason:

Currently for output artifact resolution we only consider executions with COMPLETE status. However, when cache is used, the current execution with the right argo pod name will be published with CACHED status [1]. Those should also be considered.

[1] https://github.com/tensorflow/tfx/blob/29349c9bd0196af3a7c0336a5339897406cbfdb3/tfx/components/base/base_driver.py#L263

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/3228)

### CODE REVIEW

**Comment by numerology** [comment]
/assign @Bobgy 
/assign @jingzhang36 

This is my first line of TS so please let me know if I'm not doing in the right way. Thanks!

File: frontend/src/lib/OutputArtifactLoader.ts [hunk 1]
```diff
@@ -472,7 +472,7 @@ async function getExecutionInContextWithPodName(
     return undefined; // Not found, this is expected to happen normally when there's no mlmd data.
   }
   const state = foundExecution.getPropertiesMap().get('state');
-  if (!state || state.getStringValue() !== 'complete') {
```
Comment (neuromage) [inline frontend/src/lib/OutputArtifactLoader.ts]: Maybe cleaner to write

```
if (state && state.getStringValue() === 'complete' || state.getStringValue() === 'complete')) {
  return foundExecution
}
// No valid execution found.
return undefined;
```

Comment (numerology) [reply frontend/src/lib/OutputArtifactLoader.ts]: Done.

File: frontend/src/lib/OutputArtifactLoader.ts [hunk 2]
```diff
@@ -472,10 +472,11 @@ async function getExecutionInContextWithPodName(
...
-    return undefined; // Execution doesn't have a valid state, or it has not finished.
+  if (state && (state.getStringValue() === 'complete' || state.getStringValue() === 'cached')) {
```
Comment (Bobgy) [inline frontend/src/lib/OutputArtifactLoader.ts]:
```suggestion
  if (['complete', 'cached'].includes(state?.getStringValue()) {
```
nit: can be simplified a little with some new syntax -- [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

**Comment by numerology** [comment]
/retest

Fixed the ml-metadata store image

**Comment by numerology** [comment]
> /approve
> Thanks for trying a UI fix!
> 
> @numerology Just want to double check whether you tried these changes locally. You can follow guides in https://github.com/kubeflow/pipelines/blob/master/frontend/README.md to run a local dev server and proxy api requests to your real cluster.

Thanks for the info. Will do

File: frontend/src/lib/OutputArtifactLoader.ts (hunk 2 above)
Comment (numerology) [reply frontend/src/lib/OutputArtifactLoader.ts]: Done. Thx

**Comment by numerology** [comment]
@Bobgy Did a local test. Looks good. See https://screenshot.googleplex.com/wdio46kv1P9

Thanks for the guideline!

**Comment by numerology** [comment]
/hold cancel

Verification passed.

**Pull request merged**